### ISSUE TITLE
Changes to handling of puf.csv since OG-USA 0.5.5

### ISSUE THREAD

**Issue by PeterDSteinberg** [open-source-economics/OG-USA#248]
I have had success testing OG-USA 0.5.5 with Tax-Calc 0.6.6 and 0.6.8, but I have not had success with the current (last ca 2 weeks) master branch of OG-USA and Tax-Calc 0.6.8.  Here are a few notes:

I am getting this error when using the 10.28.2016 `puf.csv`:

```
Traceback (most recent call last):
  File "run_reforms.py", line 183, in <module>
    main()
  File "run_reforms.py", line 157, in main
    run_micro=run_micro)
  File "run_reforms.py", line 87, in run_micro_macro
    runner(**kwargs)
  File "../execute.py", line 40, in runner
    start_year=user_params['start_year'], reform=reform, guid=guid)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/txfunc.py", line 1566, in get_tax_func_estimate
    analytical_mtrs, age_specific, reform)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/txfunc.py", line 1039, in tax_func_estimate
    start_year=beg_yr, reform=reform)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/get_micro_data.py", line 139, in get_data
    reform=reform)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/get_micro_data.py", line 86, in get_calculator
    records1 = Records()
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/taxcalc/records.py", line 185, in __init__
    self._read_data(data, exact_calculations)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/taxcalc/records.py", line 381, in _read_data
    raise ValueError(msg.format(varname))
ValueError: Records data variable name nu13 not in VALID_READ_VARS
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

I have confirmed I am using the `puf.csv` 10.28.2016 by using `md5sum` in a build script right before `run_reforms.py`:

`18ea2de38fcd4a22013a4b6b5af6553c`

which makes sense when I run `md5sum` locally on the file.

When I run with `puf.csv` `10.27.2016` I am getting a different puf related error:

```
Traceback (most recent call last):
  File "run_reforms.py", line 183, in <module>
    main()
  File "run_reforms.py", line 157, in main
    run_micro=run_micro)
  File "run_reforms.py", line 87, in run_micro_macro
    runner(**kwargs)
  File "../execute.py", line 40, in runner
    start_year=user_params['start_year'], reform=reform, guid=guid)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/txfunc.py", line 1566, in get_tax_func_estimate
    analytical_mtrs, age_specific, reform)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/txfunc.py", line 1039, in tax_func_estimate
    start_year=beg_yr, reform=reform)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/get_micro_data.py", line 153, in get_data
    mtr_combined_capinc = cap_inc_mtr(calc1)
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/get_micro_data.py", line 241, in cap_inc_mtr
    all_mtrs = {income_source: calc1.mtr(income_source) for income_source in capital_income_sources_taxed}
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/ogusa-0.5.5+227.gaec9d17-py2.7.egg/ogusa/get_micro_data.py", line 241, in <dictcomp>
    all_mtrs = {income_source: calc1.mtr(income_source) for income_source in capital_income_sources_taxed}
  File "/dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env/lib/python2.7/site-packages/taxcalc/calculate.py", line 302, in mtr
    raise ValueError(msg.format(variable_str))
ValueError: mtr variable_str="e26270" is not valid
Build step 'Execute shell' marked build as failure
```

**Comment by jdebacker** [comment]
@PeterDSteinberg the second error you show would suggest to me that you are using TC version prior to 0.6.8 (I believe the is the version that added `e26270` to the list of acceptable variables from which the mtr could be calculated).

Re the first error, I don't know the history of this variable.  I understand that there were some changes related to it recently, but @andersonfrailey may be better able to comment.

FYI, I was able to successfully run the OG-USA master locally with TC 0.6.8 and the PUF from 10/27.

**Comment by PeterDSteinberg** [comment]
I'm fairly certain this is with the regression branch (essentially master) of OG-USA with taxcalc 0.6.8 from conda.  In the build script I print out the output of `conda list`.  Here's what it shows:

```
RUN REFORMS
# conda environments:
#
ogusa_env             *  /dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env
root                     /dev/jenkins/workspace/ci-mode-simple-reform0/miniconda

# packages in environment at /dev/jenkins/workspace/ci-mode-simple-reform0/miniconda/envs/ogusa_env:
#
cycler                    0.10.0                   py27_0  
dateutil                  2.4.1                    py27_0  
enum34                    1.1.6                    py27_0  
funcsigs                  1.0.2                    py27_0  
libgfortran               3.0.0                         1  
libpng                    1.6.22                        0  
llvmlite                  0.11.0                   py27_0  
matplotlib                1.5.1               np110py27_0  
nomkl                     1.0                           0  
numba                     0.26.0              np110py27_0  
numpy                     1.10.4             py27_nomkl_2  [nomkl]
ogusa                     0.5.5+227.gaec9d17           <pip>
openblas                  0.2.14                        4  
openssl                   1.0.2j                        0  
pandas                    0.18.1              np110py27_0  
pip                       9.0.0                    py27_0  
pyparsing                 2.1.4                    py27_0  
pytest                    3.0.3                    py27_0  
python                    2.7.12                        1  
pytz                      2016.7                   py27_0  
readline                  6.2                           2  
scipy                     0.17.1          np110py27_nomkl_1  [nomkl]
setuptools                27.2.0                   py27_0  
singledispatch            3.4.0.3                  py27_0  
six                       1.10.0                   py27_0  
sqlite                    3.13.0                        0  
taxcalc                   0.6.8               np110py27_2    ospc
tk                        8.5.18                        0  
toolz                     0.8.0                    py27_0  
wheel                     0.29.0                   py27_0  
yaml                      0.1.6                         0  
zlib                      1.2.8                         3 
```

**Comment by jdebacker** [comment]
This issue has been fixed with PR #324 and others.
